pkg/kapis/devops/v1alpha2/devops.go

```diff
@@ -41,15 +44,81 @@ func (h *ProjectPipelineHandler) GetPipeline(req *restful.Request, resp *restful
 	resp.WriteAsJson(res)
 }
 
+func (h *ProjectPipelineHandler) getPipelinesByRequest(req *restful.Request) (api.ListResult, error) {
+	// this is a very trick way, but don't have a better solution for now
+	var (
+		err       error
+		start     int
+		limit     int
+		namespace string
+	)
+
+	// parse query from the request
+	query := req.QueryParameter("q")
+	req.Request.PostForm.Set("limit", "1000")
+	for _, val := range strings.Split(query, ";") {
+		if strings.HasPrefix(val, "pipeline:") {
+			namespace = strings.TrimLeft(val, "pipeline:")
+			namespace = strings.Split(namespace, "/")[0]
+		}
+	}
+
+	// make sure we have an appropriate value
+	if start, err = strconv.Atoi(req.QueryParameter("start")); err != nil {
+		start = 0
+	}
+	if limit, err = strconv.Atoi(req.QueryParameter("limit")); err != nil {
+		limit = 10
+	}
+
+	defer req.Request.Form.Set("limit", "10000") // assume the pipelines no more than 10k
```

Comment:
get the value of the limit from QueryParmeter or set it as 10.  What's the intention here?

Reply:
There are two lists here. And they might have a different order. So this list must be larger than another one. Anyway, the front-end will get the correct data.

Reply:
It makes me a little confused. Why does it need to set req.Request.Form and req.Request.PostForm at the same time?

Reply:
[The previous one](https://github.com/kubesphere/kubesphere/pull/3149/files#diff-38024724898ee9b3e4ca7f645d4c19869b100c394ccf587ef1e1ea554f3a7a2aR58) should be moved. It's useless.

Reply:
fixed it https://github.com/kubesphere/kubesphere/pull/3149/commits/62c297ec53049f2e08e873ea1ee5aa3412b54d5e#diff-38024724898ee9b3e4ca7f645d4c19869b100c394ccf587ef1e1ea554f3a7a2aL58